pkgs/tools/security/pinentry-mac/default.nix

```diff
@@ -17,6 +17,10 @@ stdenv.mkDerivation rec {
     mv build/Release/pinentry-mac.app $out/Applications
   '';
 
+  passthru = {
+    binaryPath = "Applications/pinentry-mac.app/Contents/MacOS/pinentry-mac";
```

Comment:
You probably want to add ```binaryPath``` to the normal pinentry as well and get rid of the ```or "bin/pinentry"``` clause